app/javascript/widget/components/Form/PhoneInput.vue

```diff
@@ -0,0 +1,314 @@
+import FluentIcon from 'shared/components/FluentIcon/Index.vue';
+import {
+  hasPressedArrowUpKey,
+  hasPressedArrowDownKey,
+  isEnter,
+} from 'shared/helpers/KeyboardHelpers';
+import FormulateInputMixin from '@braid/vue-formulate/src/FormulateInputMixin';
+import darkModeMixin from 'widget/mixins/darkModeMixin';
+
+export default {
+  components: {
+    FluentIcon,
+  },
+  mixins: [eventListenerMixins, FormulateInputMixin, darkModeMixin, clickaway],
+  props: {
+    placeholder: {
+      type: String,
+      default: '',
+    },
+    hasErrorInPhoneInput: {
+      type: Boolean,
+      default: false,
+    },
+  },
+  data() {
+    return {
+      countries: [
+        {
+          name: 'Select Country',
+          dial_code: '',
+          emoji: '',
+          id: '',
+        },
+        ...countries,
+      ],
+      selectedIndex: -1,
+      showDropdown: false,
+      searchCountry: '',
+      activeCountryCode: '',
+      activeDialCode: '',
+      phoneNumber: '',
+    };
+  },
+  computed: {
+    dropdownClass() {
+      return `${this.$dm('bg-slate-100', 'dark:bg-slate-700')} ${this.$dm(
+        'text-slate-700',
+        'dark:text-slate-50'
+      )}`;
+    },
+    dropdownBackgroundClass() {
+      return `${this.$dm('bg-white', 'dark:bg-slate-700')} ${this.$dm(
+        'text-slate-700',
+        'dark:text-slate-50'
+      )}`;
+    },
+    dropdownItemClass() {
+      return `${this.$dm('text-slate-700', 'dark:text-slate-50')} ${this.$dm(
+        'hover:bg-slate-50',
+        'dark:hover:bg-slate-600'
+      )}`;
+    },
+    activeDropdownItemClass() {
+      return `active ${this.$dm('bg-slate-100', 'dark:bg-slate-800')}`;
+    },
+    focusedDropdownItemClass() {
+      return `focus ${this.$dm('bg-slate-50', 'dark:bg-slate-600')}`;
+    },
+    inputHasError() {
+      return this.hasErrorInPhoneInput
+        ? `border-red-200 hover:border-red-300 focus:border-red-300 ${this.inputLightAndDarkModeColor}`
+        : `hover:border-black-300 focus:border-black-300 ${this.inputLightAndDarkModeColor} ${this.inputBorderColor}`;
+    },
+    inputBorderColor() {
+      return `${this.$dm('border-black-200', 'dark:border-black-500')}`;
+    },
+    inputLightAndDarkModeColor() {
+      return `${this.$dm('bg-white', 'dark:bg-slate-600')} ${this.$dm(
+        'text-slate-700',
+        'dark:text-slate-50'
+      )}`;
+    },
+    filteredCountriesBySearch() {
+      return this.countries.filter(country => {
+        const { name, dial_code, id } = country;
+        const search = this.searchCountry.toLowerCase();
+        return (
+          name.toLowerCase().includes(search) ||
+          dial_code.toLowerCase().includes(search) ||
+          id.toLowerCase().includes(search)
+        );
+      });
+    },
+    activeCountry() {
+      if (this.activeCountryCode) {
+        return this.countries.find(
+          country => country.id === this.activeCountryCode
+        );
+      }
+      return '';
```

Comment:
This could be simplified to 
```suggestion
      return this.countries.find(
          country => country.id === this.activeCountryCode
        ) || '';
```